Documentation/kube-aws-approver.md

```diff
@@ -0,0 +1,100 @@
+# Kubernetes AWS approver
+
+A custom node CertificateSigningRequest approver for kubernetes clusters running on AWS.
+
+## Why do I want this?
+
+[kubelet TLS bootstrapping][tls-bootstrapping] allows kubelet to create CertificateSigningRequest for its client certs. These CertificateSigningRequest need to be approved so as to be signed by kubernetes internal certificate signer controller.
+
+Currently you can either manually approve each node that tries to join the cluster. Or, instruct kubernetes internal certificate approver to approve all certificates from `system:bootstrappers` group. The human intervention required for manual approval is not ideal in auto scaling environments. And the default auto approver performs very basic checks that might not be very secure.
+
+`kube-aws-approver` goes a step further from default auto approver and uses the CertificateSigningRequest generated by node to extract information about the node requesting the certificate, and tries to validate it against AWS API before approving the CertificateSigningRequest.
```

Comment:
Please add a link for the `AWS API`

Reply:
I used `AWS API` in a general sense. Not a particular api endpoint.